packages/backend/src/database/migrations/20241128121723_add-metric-tree-edges-table.ts

```diff
@@ -0,0 +1,33 @@
+import { Knex } from 'knex';
+
+const METRICS_TREE_EDGES_TABLE = 'metrics_tree_edges';
+const CATALOG_SEARCH_TABLE = 'catalog_search';
+
+export async function up(knex: Knex): Promise<void> {
+    if (!(await knex.schema.hasTable(METRICS_TREE_EDGES_TABLE))) {
+        await knex.schema.createTable(METRICS_TREE_EDGES_TABLE, (table) => {
+            table
+                .uuid('source_catalog_search_uuid')
+                .notNullable()
+                .references('catalog_search_uuid')
+                .inTable(CATALOG_SEARCH_TABLE)
+                .onDelete('CASCADE');
+
+            table
+                .uuid('target_catalog_search_uuid')
+                .notNullable()
+                .references('catalog_search_uuid')
+                .inTable(CATALOG_SEARCH_TABLE)
+                .onDelete('CASCADE');
+
+            table.primary([
+                'source_catalog_search_uuid',
+                'target_catalog_search_uuid',
+            ]);
+        });
+    }
```

Comment:
does this table gets refreshed every time catalog refreshes? I guess yes but im still conflicted with the idea of using catalog search uuids in any form and just rely on (table, metric) pairs as primary source of links

Reply:
I completely agree, I think we should get rid of that ID and use `tableName`, `name` as the primary key, but for now we gotta use these UUIDs

Not sure when we should do this tbh